discopy/cat.py

```diff
@@ -722,12 +721,11 @@ def __iter__(self):
     def __len__(self):
         return len(self.terms)
 
-    def then(self, *others):
-        if len(others) != 1:
-            return super().then(*others)
-        other = others[0] if isinstance(others[0], Sum) else Sum(list(others))
+    @unbiased
+    def then(self, other):
+        other = other if isinstance(other, Sum) else Sum((other, ))
         unit = Sum([], self.dom, other.cod)
-        terms = [f.then(g) for f in self.terms for g in other.terms]
+        terms = tuple(f.then(g) for f in self.terms for g in other.terms)
```

Comment:
```suggestion
terms = [f.then(g) for f in self.terms for g in other.terms]
```

In this version the terms of a sum should be a list rather than a tuple, in the init it will copy everything back to a list:
https://github.com/discopy/discopy/blob/b93b52d54af29944ea610530bab8bdc94f632f6a/discopy/cat.py#L674